lib/plugins/keyGeneratorFactory.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+// TODO: should be instance! cannot use Nocca instance references now.
+
+var $q = require('q');
+var $extend = require('extend');
+
+module.exports = {
+    interface: 'keyGenerator',
+    id: 'keyGeneratorFactory',
+    name: 'Key Generator Factory',
+    constructor: KeyGeneratorFactory
+};
+
+function KeyGeneratorFactory (Nocca) {
+
+    var self = this;
+    
+    self.generateKey = generateKey;
+
+    self.incrementalKeyGenerator = incrementalKeyGenerator;
```

Comment:
Should we split this in multiple keyGenerator plugins ?? 

Then we'd have to allow for the combination of Generators from Nocca natively maybe. You could then potentially combine any keyGenerator-plugins to get your key.